### PR TITLE
refactor(conda): Use format strings

### PR DESCRIPTION
**Pull request by ATiltedTree**
#### Description

This PR deprecates `symbol` and `style` keys in the `conda` module to replace it with the `format` key instead.
It also changes the default symbol to the reference implementation.

#### Motivation and Context

Related to issue #1057

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):
image

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: tests/testsuite/conda.rs [hunk 1]
```diff
@@ -2,15 +2,10 @@ use ansi_term::Color;
...
-use crate::common::TestCommand;
 
 #[test]
 fn not_in_env() -> io::Result<()> {
-    let output = common::render_module("conda")
-        .env_clear()
-        .env("PATH", env!("PATH"))
-        .use_config("".parse().unwrap())
-        .output()?;
+    let output = common::render_module("conda").output()?;
```
Comment (heyrict) [inline tests/testsuite/conda.rs:8]: Hmm, is this change required? I'm wondering if removing `.env_clear()` will make the development environment messed up with the tests.

Comment (ATiltedTree) [reply tests/testsuite/conda.rs:8]: `.env_clear()` is actually already called by `common::render_module()`.
https://github.com/starship/starship/blob/77c42dca734b9ee537c5c6351d1422bb025b2d30/tests/testsuite/common.rs#L32-L44
